com.microsoft.mrtk.input/Interactors/GazePinch/GazePinchInteractor.cs

```diff
@@ -38,7 +38,22 @@ public class GazePinchInteractor :
         /// <summary>
         /// The worldspace pose of the hand pinching point.
         /// </summary>
-        public Pose PinchPose => handController.PinchSelectPose;
+        public Pose PinchPose => pinchPoseSource.TryGetPose(out Pose pinchPose) ? pinchPose : Pose.identity;
+
+        [SerializeReference]
+        [InterfaceSelector]
+        [Tooltip("The pose source representing the ray this interactor uses for rotation")]
+        private IPoseSource rayPoseSource;
```

Comment:
The naming here feels a little odd, since this isn't really used to get the _ray's_ pose, but instead is used to get the device pose for manipulation after a selection.

These changes will also eventually lead to a comment below on line 214 becoming incorrect, but I think for now it's still accurate:

> // Get the actual device/grab rotation. The controller transform is the aiming pose;
> // we must get the underlying grab rotation.

Reply:
The way it was described, the device's pose right now is bound to the `position` and `rotation` input action, which is the "rays'" position and rotation. Thus the naming.

Reply:
The device's pose (grip pose) is distinct from the pointing ray pose (aim pose), as described in the OpenXR spec: https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#semantic-path-standard-pose-identifiers.

My understanding is that we use the grip pose here to get full orientation data for the controller/hand doing the interacting. That's at least what the old code was doing (`device.deviceRotation.ReadValue()`), which is reading the underlying device's pose directly and not the (currently) mapped pointing pose input action.